fitparser/src/de/parser.rs

```diff
@@ -210,8 +211,7 @@ pub struct DeveloperFieldDefinition {
 pub struct FitDataMessage {
     global_message_number: u16,
     time_offset: Option<u8>,
-    fields: HashMap<u8, Value>,
-    developer_fields: Vec<Value>,
+    fields: HashMap<(u8, u8), Value>,
```

Comment:
I think I'd rather keep the non-developer field data separate here and still have the "regular" fields stored as a hash map of `HashMap<u8, Value>`. Parsing the normal fields is the still the main execution path so it would be ideal to not change that part of the public API.

Storing developer fields as `HashMap<(u8, u8), Value>` is fine, I can see where we might not want to assume there's only a single developer ID present in the file

Reply:
I have pushed a commit that does this, see below for one issue that this causes with the borrow checker